internal/controller/authenticator/jwtcachefiller/jwtcachefiller.go

```diff
@@ -125,20 +171,54 @@ func (c *controller) Sync(ctx controllerlib.Context) error {
 		}
 	}
 
+	conditions := make([]*metav1.Condition, 0)
+	specCopy := obj.Spec.DeepCopy()
+	var errs []error
+
+	rootCAs, conditions, tlsOk := c.validateTLS(specCopy.TLS, conditions)
+	_, conditions, issuerOk := c.validateIssuer(specCopy.Issuer, conditions)
+
+	client := phttp.Default(rootCAs)
+	client.Timeout = 30 * time.Second // copied from Kube OIDC code
+	coreOSCtx := coreosoidc.ClientContext(context.Background(), client)
+
+	pJSON, provider, conditions, providerErr := c.validateProviderDiscovery(coreOSCtx, specCopy.Issuer, conditions, tlsOk && issuerOk)
+	errs = append(errs, providerErr)
+
+	jwksURL, conditions, jwksErr := c.validateProviderJWKSURL(provider, pJSON, conditions, tlsOk && issuerOk && providerErr == nil)
+	errs = append(errs, jwksErr)
+
+	keySet, conditions, jwksFetchErr := c.validateJWKSFetch(coreOSCtx, jwksURL, conditions, tlsOk && issuerOk && providerErr == nil && jwksErr == nil)
+	errs = append(errs, jwksFetchErr)
```

Comment:
How about something like this? (check this for typos too because I'm typing this without a compiler to help)

```suggestion
	rootCAs, conditions, tlsOk := c.validateTLS(specCopy.TLS, conditions)
	_, conditions, issuerOk := c.validateIssuer(specCopy.Issuer, conditions)
        okSoFar := tlsOk && issuerOk

	client := phttp.Default(rootCAs)
	client.Timeout = 30 * time.Second // copied from Kube OIDC code
	coreOSCtx := coreosoidc.ClientContext(context.Background(), client)

	pJSON, provider, conditions, providerErr := c.validateProviderDiscovery(coreOSCtx, specCopy.Issuer, conditions, okSoFar)
	errs = append(errs, providerErr)
	okSoFar = okSoFar && providerErr == nil

	jwksURL, conditions, jwksErr := c.validateProviderJWKSURL(provider, pJSON, conditions, okSoFar)
	errs = append(errs, jwksErr)
	okSoFar = okSoFar && jwksErr == nil

	keySet, conditions, jwksFetchErr := c.validateJWKSFetch(coreOSCtx, jwksURL, conditions, okSoFar)
	errs = append(errs, jwksFetchErr)
```

Reply:
...and continue that pattern for the rest of the function